wordpress-mobile/WordPress-Android
# Limit Classic editor access on Simple WP.com sites to existing Classic posts

The next step after [notifying users that the classic editor is going away](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3209) is to actually remove it, starting with Simple sites.

I think this change doesn't warrant inclusion in RELEASE-NOTES.txt [since we already notified users about this change in 16.9](https://github.com/wordpress-mobile/WordPress-iOS/pull/16008), but if anyone thinks differently I'd love to hear. Perhaps we could include it in release notes but exclude it from the Play Store version notes.

Addresses https://github.com/wordpress-mobile/gutenberg-mobile/issues/3092 (for WPAndroid)



## To test

_Note: The toggle switch referred to below is the "Use block editor" toggle switch found in My Site → Settings. This is only available to site admins._

1. With APK from this PR, log into a WP.com account
2. Choose a Simple site where you are an admin
3. Verify that there **is no** toggle switch
4. Using the current Play Store version of the app – not the app built from this branch – log into the same WP.com account
5. Choose the same Simple site where you are an admin, locate the toggle switch and turn it off (meaning use the classic editor) 
6. Moving back over to the app built from this branch – locate the same Simple site
7. Create a new post and ensure that the block editor is shown (not the classic editor)

## Regression Notes
1. Potential unintended areas of impact

This PR removes the classic editor on Simple sites. However, we're leaving it in place for scenarios where the block editor cannot be used instead. These scenarios include editing a classic post (i.e. a post that contains only classic content). We have to check to make sure these scenarios still work on Simple sites. We should also check to make sure all other site types (Atomic/Jetpack/self-hosted) are unaffected by this change.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- [x] Test Simple WP.com sites do not show a toggle switch (used camiams.home.blog)
- [x] Test Simple WP.com sites defaulted on mobile to Classic use Gutenberg for new posts (used camiams.home.blog)
- [x] Regression test Atomic sites show a toggle switch (https://americanswimming9.wpcomstaging.com/)
- [x] Regression test Jetpack-connected self-hosted sites show a toggle switch (used coderovers.io)
- [x] Regression test self-hosted sites show a toggle switch (used codrovers.io)
- [x] Regression test Simple WP.com site users open existing Classic posts in the Classic editor (camiams.home.blog)

3. What automated tests I added (or what prevented me from doing so)

We are disabling the Classic Editor tests until they can be migrated to executing on existing classic posts, instead of creating new ones. That work is tracked with this ticket here: https://github.com/wordpress-mobile/WordPress-Android/issues/14389

PR submission checklist:
